### ISSUE TITLE
Hyper-Diffusion: Estimating Epistemic and Aleatoric Uncertainty with a Single Model

### ISSUE THREAD

**Issue by wrkhard** [lightning-uq-box/lightning-uq-box#164]
Hi Nils,

I recently came across this paper and thought of the work you've done on trying to democratize CARDS. This looks like perhaps a nice extension: https://arxiv.org/abs/2402.03478 

Might be a nice addition to your package.

Best,
-Will

**Comment by nilsleh** [comment]
Hi, thanks for the recommendation, that does look interesting. Wondering if it can be integrated with the DDPM #105 approach I am trying to implement. But also looks like they have not published any code :/

**Comment by wrkhard** [comment]
Yes, I was also disheartened to find no public code available yet and plan
on contacting the author. To my understanding implementation may be
somewhat straight forward. I have been modifying your CARD
implementation to also provide epistemic uncertainty through a simple
ensemble (once I optimize out the kinks I would be more than happy to
submit this to your codebase!) But the notion of the Hypernetwork indeed
might be much more useful for problems with higher compute constraints..

**Comment by nilsleh** [comment]
Yeah would be great if you want to contribute that, looking forward to your PR. Probably best to discuss the Hypernetwork idea then?

**Comment by wrkhard** [comment]
Hi Nils,

Sorry for the slow reply. I contacted the authors and it turns out the
paper is preprint but hopefully will be accepted soon. They plan on
releasing code once that happens. I will try to get my small contribution
pushed soon on CARD. Happy to discuss the Hypernetwork further as well.
Seems like there are some papers that utilize the Hypernetwork as a means
of expanding on UQ from Deep Ensembles so might be a worthwhile endeavor!

Best,
-Will

**Comment by wrkhard** [comment]
Hi Nils,

If there is still interest in this a repo has recently been released along side the published paper.

Paper: https://arxiv.org/pdf/2402.03478
Repo: https://github.com/matthewachan/hyperdm
